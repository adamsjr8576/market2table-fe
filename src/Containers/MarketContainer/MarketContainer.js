import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Route, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './MarketContainer.scss';
import { Market } from '../../Components/Market/Market';

export const MarketContainer = () => {

  const { markets } = useSelector(state => state);
  const displayMarkets = markets.map(market => {
    return (
      <Market
        key={market.id}
        id={market.id}
        name={market.name}
      />
    )
  });
  let display;
  if (markets.length) {
    display = displayMarkets;
  } else {
    display = (<p>LOADING....</p>);
  }
  return (
    <section className='section-market-container'>
      {display}
    </section>
  )
}

export default MarketContainer;

MarketContainer.propTypes = {
  markets: PropTypes.array,
}
