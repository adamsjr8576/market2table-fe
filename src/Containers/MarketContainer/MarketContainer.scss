.section-market-container {
  width: 25%;
  height: 100%;
}
