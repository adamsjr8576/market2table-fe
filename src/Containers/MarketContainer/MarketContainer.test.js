import React from 'react';
import { shallow } from 'enzyme';
import { MarketContainer } from './MarketContainer';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
  markets: [{name: 'Friendly Famrs', id:'1923782', image_link: 'image.png'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState
}));

describe('MarketContainer', () => {

  let wrapper;

  it('should match the snapshot with markets', () => {
    wrapper = shallow(<MarketContainer
      />);

    expect(wrapper).toMatchSnapshot();
  });
});
