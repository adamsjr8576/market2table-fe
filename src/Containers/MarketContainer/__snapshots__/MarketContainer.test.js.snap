// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketContainer should match the snapshot with markets 1`] = `
<section
  className="section-market-container"
>
  <Market
    id="1923782"
    key="1923782"
    name="Friendly Famrs"
  />
</section>
`;
